### DISCUSSION TITLE
Can't load remote image source - Blocked by CORS policy - Access-Control-Allow-Origin

### DISCUSSION THREAD

**Discussion by cspecter** [directus/directus#15090] [Archive]
### Describe the Bug

I have created a very simple extension that creates a URL to an external image that I want to display in the content form. The url is just passed to the `:src` on an `<img>` tag. This worked fine in 9.6.0, but since upgrading to 9.15.1 I am getting the following error when it tries to fetch the image.

It does not appear to be sending the correct headers, instead sending
```
Accept: application/json, text/plain, */*
```

I have tried to add CORS settings to the `.env`, but nothing I put in there is having any affect on this issue.
```
##  CORS
CORS_ENABLED=true
CORS_ORIGIN="array:https://localhost,https://axzdfdpwfsynrajqqoae.supabase.co" // Also tried '*', 'http' and just the domain
CORS_METHODS=GET,POST,PATCH,DELETE,OPTIONS
MAX_PAYLOAD_SIZE="10mb"
CORS_CREDENTIALS=true
CORS_ALLOWED_HEADERS=Accept,X-Request-With,Content-Type,Authorization,Origin
```

Additionally I have tried to set the following variables in the `.env`:
`CONTENT_SECURITY_POLICY_DIRECTIVES__UPGRADE_INSECURE_REQUESTS=null`  - Does nothing to fix the problem
`CONTENT_SECURITY_POLICY_USE_DEFAULTS=false` - Directus will not start when this is set.

Any idea why this may be happening?

### To Reproduce

You can build and install my extension here:

https://github.com/cspecter/directus-extensions/tree/master/packages/directus-m2o-image-display

This is pretty simple, it's taking the response from the DB, say a `path` key and appending that to a url-prefix. You should be able to just put a whole URL to an external image in the URL-Prefix form. 

### Errors Shown

```
Access to XMLHttpRequest at 'https://axzdfdpwfsynrajqqoae.supabase.co/storage/v1/object/public/files/products/cover/vlcsnap-2022-05-05-11h51m04s253_g10_05052022.jpg'
from origin 'https://cms.gethybrid.co' has been blocked by CORS policy: Response to preflight request doesn't pass
access control check: The value of the 'Access-Control-Allow-Origin' header in the response must not be the
wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the
XMLHttpRequest is controlled by the withCredentials attribute.
```

```
Refused to load the image '<URL>' because it violates the following Content Security Policy directive: "img-src 'self' data: blob: <URL>".
```

### What version of Directus are you using?

9.15.1

### What version of Node.js are you using?

12.0.0

### What database are you using?

Postgres 14.5

### What browser are you using?

Chrome

### How are you deploying Directus?

Digital Ocean instance

**Comment by rijkvanzanten** [comment]
By default, the app's content security policy headers only allow images from self, inline data/blob, and our know safe cdn:

https://github.com/directus/directus/blob/9c0f806ed832c0491e43db09daf2f4ce4197008e/api/src/app.ts#L120

You can update that array through the `CONTENT_SECURITY_POLICY__IMG_SRC=""` environment variable 👍🏻

**Reply by cspecter** [reply]
This worked for me. Thanks!

I know you mention this in the docs, as part of the helmet setup, but you might want document this in a more full way. It's a very confusing subject if you don't know where to look and a frustrating problem to deal with.
